include/deal.II/lac/constraint_linear_operator.h

```diff
@@ -0,0 +1,291 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__constraint_linear_operator_h
+#define dealii__constraint_linear_operator_h
+
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/constraint_matrix.h>
+
+#ifdef DEAL_II_WITH_CXX11
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/**
+ * @name Indirectly applying constraints to LinearOperator
+ */
+//@{
+
+
+/**
+ * This function takes a ConstraintMatrix @p cm and an operator exemplar @p
+ * exemplar (this exemplar is usually a linear operator that describes the
+ * system matrix) and returns a LinearOperator object associated with the
+ * "homogeneous action" of the underlying ConstraintMatrix object:
+ *
+ * Applying the LinearOperator object on a vector <code>u</code> results in
+ * a vector <code>v</code> that stores the result of calling
+ * ConstraintMatrix::distribute() on <code>u</code> - with one important
+ * difference; inhomogeneities are note applied, but always treated as 0
```

Comment:
I think it would actually be useful to state what the operator does for each non-constrained/constrained element of the vector.